eclair-core/src/main/scala/fr/acinq/eclair/blockchain/electrum/ElectrumClientPool.scala

```diff
@@ -119,15 +119,15 @@ class ElectrumClientPool(serverAddresses: Set[InetSocketAddress])(implicit val e
 
   initialize()
 
-  private def handleHeader(connection: ActorRef, tip: ElectrumClient.Header, d: Option[ConnectedData]) = {
+  private def handleHeader(connection: ActorRef, tip: ElectrumClient.Header, data: Option[ConnectedData]) = {
     // we update our block count even if it doesn't come from our current master
     updateBlockCount(tip.block_height)
-    d match {
+    data match {
       case None =>
         // as soon as we have a connection to an electrum server, we select it as master
         log.info(s"selecting master ${addresses(connection)} at $tip")
-        statusListeners.foreach(_ ! ElectrumClient.ElectrumReady(tip))
-        context.system.eventStream.publish(ElectrumClient.ElectrumReady(tip))
+        statusListeners.foreach(_ ! ElectrumClient.ElectrumReady(tip, addresses(connection)))
```

Comment:
nit: Maybe worth assigning `val remoteAddress = addresses(connection)` at the top of the method.